### ISSUE TITLE
Spec with a filter with timeUnit cannot be rendered

### ISSUE THREAD

**Issue by starry97** [vega/vega-lite#2756]
The following spec cannot be rendered.

```
{
  "data": {
    "url": "node_modules/vega-datasets/data/cars.json"
  },
  "transform": [
    {
      "filter": {
        "field": "Year",
        "timeUnit": "year",
        "range": [
          1969,
          1981
        ]
      }
    }
  ],
  "mark": "point",
  "encoding": {
    "y": {
      "field": "Origin",
      "type": "nominal"
    }
  },
  "config": {
    "overlay": {
      "line": true
    },
    "scale": {
      "useUnaggregatedDomain": true
    }
  }
}
```

**Comment by kanitw** [comment]
Yeah this is indeed a bug in Vega-Lite, we need to parse `timeUnit` from `filter`.

**Comment by starry97** [comment]
```
{
  "data": {
    "url": "node_modules/vega-datasets/data/cars.json"
  },
  "transform": [
    {
      "filter": {
        "field": "Year",
        "timeUnit": "yearmonthdate",
        "range": [
          {
            "year": 1969,
            "quarter": 4,
            "month": 12,
            "date": 31,
            "day": 4,
            "hours": 16,
            "minutes": 0,
            "seconds": 0,
            "milliseconds": 0,
            "utc": false
          },
          {
            "year": 1981,
            "quarter": 4,
            "month": 12,
            "date": 31,
            "day": 5,
            "hours": 16,
            "minutes": 0,
            "seconds": 0,
            "milliseconds": 0,
            "utc": false
          }
        ]
      }
    }
  ],
  "mark": "point",
  "encoding": {
    "y": {
      "field": "Cylinders",
      "type": "nominal"
    }
  },
  "config": {
    "overlay": {
      "line": true
    },
    "scale": {
      "useUnaggregatedDomain": true
    }
  }
}
```

**Comment by kanitw** [comment]
For now, a work around is to add format parse to data

```
"data": {
    "url": "data/cars.json",
    "format": {
      "parse": {
        "Year": "date"  
      }
    }
  }
```

**Comment by kanitw** [comment]
Fixed in #2805
